docs/proposals/docs-process.md

```diff
@@ -0,0 +1,94 @@
+* Reduce dependency on scripts and tools that only a few people can use/understand
+
+Cost:
+
+* Migration of current head/stable release system that is done with build/integration scripts into GitHub branches
+* Replacing of doc-munging scripts with native Jekyll functionality and/or GitHub capability.
+
+# Github.io staging
+
+GitHub has a technology called GitHub Pages that we currently use to construct the site at `kubernetes.io`. GitHub Pages is essentially a system for managing and hosting Jekyll sites. When you check in a Jekyll site into a repo that is named `username.github.io`, then GitHub will automatically stage its contents at `http://username.github.io`, and automatically update it whenever you push new changes -- no publishing required.
+
+This highly lubricates the ability for external users to make changes, as they can immediately see the consequences of their actions in their forked environment. 
+
+## Why migrate to kubernetes.github.io instead of using a `gh-pages` branch?
+
+GitHub Pages also suppots using branches of an existing repo to host Jekyll sites; in this case you would reuse the repo, but check in the Jekyll site to a branch named `gh-pages`. While you still get auto-building and staging, it will build at `http://username.github.io/**repo**` -- and in our case that is problematic because absolute links to any URI within the site will not work the same as they will on kubernetes.io. If we use the `username.github.io` repo instead of the gh-pages branch, then including an image from `/images/example.png` will work the same on `http://kubernetes.github.io`, `http://your_username.github.io`, `http://kubernetes.io`, and your local machine (running `jekyll serve` off of a clone of our repo).
+
+## Overview of docs migration and how release process will change
+
+Right now, the subdirectory of kubernetes/docs acts as the HEAD for the docs, which is used to populate new releases, as well as the basis for cherrypicking changes into docs for old releases. However, consider a "roaming HEAD" situation, such that the new release is always treated as the basis for populating new releases and  cherrypicks. In otherwords, whatever information is `http://kubernetes.io/vNEWEST/` is HEAD, and we no longer have a dependency on an underlying “master” docset. 
+
+New releases can be developed under `http://kubernetets.io/vNEXT`. We can do this because there is an extensive set of steps to get any one particular folder full of docs to be the default. Or, at worst, we can make a vNEXT branch of the `kubernetes/kubernetes.github.io` repo and do vNEXT work there, culminating in a pull request at release time. But this would only be necessary if we don’t want users to ever see what vNEXT docs look like. I personally vote for doing our work out in public. We can always mark those docs "beta" -- and keeping them in the master branch means that we get to use all the great github.io staging tools. 
```

Comment:
For me the biggest advantage of docs being in the master branch is that we can demand coordinated commitment of features and docs.  Having them in different branches seems like an invitation to drift.

To be clear, I am not against this plan, per se, just explaining my thoughts on the tradeoffs.